hawkular-inventory-impl-tinkerpop-parent/hawkular-inventory-impl-tinkerpop-sql-provider/src/main/java/org/hawkular/inventory/impl/tinkerpop/sql/SqlGraphProvider.java

```diff
@@ -144,4 +146,16 @@ private static Object sampleValue(Class<?> type) {
             throw new IllegalArgumentException("Unhandled type of property: " + type);
         }
     }
+
+    @Override public RuntimeException translateException(RuntimeException inputException, CanonicalPath affectedPath) {
+        // Sqlg sends IllegalArgumentException when unique constraint is violated
+        // See SqlgElement.updateRow
+        if (inputException instanceof IllegalArgumentException
```

Comment:
Is there no way of throwing that EAEE earlier so that this 'complicated' logic is not needed?